test/jdk/java/lang/invoke/lambda/LogGeneratedClassesTest.java

```diff
@@ -230,35 +231,33 @@ public void testDumpDirNotWritable() throws IOException {
                 return;
             }
 
-            TestResult tr = doExec(JAVA_CMD.getAbsolutePath(),
-                                   "-cp", "..",
-                                   "-Duser.dir=" + testDir.toAbsolutePath(),
+            ProcessBuilder pb = createLimitedTestJavaProcessBuilder(
+                        "-cp", CLASSES.toString(),
                                    "-Djava.security.manager=allow",
                                    "-Djdk.invoke.LambdaMetafactory.dumpProxyClassFiles",
-                                   "com.example.TestLambda");
-            assertEquals(tr.testOutput.stream()
-                                      .filter(s -> s.contains("DUMP_LAMBDA_PROXY_CLASS_FILES is not writable"))
-                                      .count(),
-                         1, "only show error once");
-            assertTrue(tr.exitValue != 0);
+                                   "com.example.TestLambda").directory(testDir.toFile());
+            executeProcess(pb)
+                    .shouldContain("DUMP_LAMBDA_PROXY_CLASS_FILES is not writable")
+                    .shouldNotHaveExitValue(0);
```

Comment:
The old code also seems to test that the error is shown exactly once (and not multiple times). Not sure if that is important to retain?

Reply:
I think that check is unnecessary.  This fails when `InnerClassLambdaMetafactory` creates the `ClassFileDumper` for lambda proxy.  It happens once.